packages/react-experiments/src/components/FloatingSuggestionsComposite/FloatingSuggestionsList/FloatingSuggestionsList.tsx

```diff
@@ -20,7 +21,44 @@ export const FloatingSuggestionsList = <T extends {}>(props: IFloatingSuggestion
   };
 
   const renderHeader = (): JSX.Element | null => {
-    const { onRenderHeader, suggestionsHeaderText } = props;
+    const {
+      onRenderHeader,
+      suggestionsHeaderText,
+      headerItemsProps,
+      selectedHeaderIndex,
+      suggestionsHeaderContainerAriaLabel,
+    } = props;
+
+    if (headerItemsProps) {
+      return (
+        <div
+          className={css('ms-Suggestions-headerContainer', classNames.suggestionsContainer)}
+          id="suggestionHeader-list"
+          role="list"
+          aria-label={suggestionsHeaderContainerAriaLabel}
+        >
+          {headerItemsProps.map((headerItemProps: ISuggestionsHeaderFooterProps, index: number) => {
+            const isSelected = selectedHeaderIndex !== -1 && selectedHeaderIndex === index;
+            return headerItemProps.shouldShow() ? (
+              <div
+                id={'sug-header' + index}
+                key={'sug-header' + index}
+                role="listitem"
+                aria-label={headerItemProps.ariaLabel}
+              >
+                <SuggestionsHeaderFooterItem
+                  id={'sug-header-item' + index}
+                  isSelected={isSelected}
+                  renderItem={headerItemProps.renderItem}
+                  onExecute={headerItemProps.onExecute}
+                  className={headerItemProps.className}
+                />
+              </div>
+            ) : null;
+          })}
+        </div>
+      );
+    }
 
     if (onRenderHeader) {
```

Comment:
should headerItemsProps and onRenderHeader checks be in an if-else if, the reason could be that we want to only show onRenderHeader when we dont have headerItemProps. If the consumer decided to provide both, in this case, i think onRenderHeader would take precendence right?

Reply:
headerItemProps would take precedence as it's first and there's a return. I added a note about how the header or footer props win over onRenderFooter/Header somewhere